packages/venia-concept/src/util/makeUrl.js

```diff
@@ -1,64 +1,77 @@
-
-// `makeResizedUrl("/path/to/jpeg", { width: 80 })` returns
-// "/img/resize/80?url=%2Fpath%2fto%2fjpeg"
-function makeResizedUrl(url, width) {
-    return `${joinUrls(resizeBase, width.toString())}?url=${encodeURIComponent(
-        url
-    )}`;
-}
+            '/media/catalog/category'
+    );
 
-// Should fail silently on a null value, so default to the empty string
-// so that string methods work
-function formatUrl(url = '', opts = {}) {
-    const { type, width } = opts;
-    if (absoluteUrl.test(url) || !(type || width)) {
-        // Absolute URLs shouldn't be resized
-        // Without a type or a width, we don't know how to transform this url
+const prependMediaPath = (url, type) => {
+    // only prepend to relative URLs of recognized type
+    if (absoluteUrl.test(url) || !type) {
         return url;
     }
 
-    // we need to throw on an unrecognized `type`, so we may have to assign this
-    // in another code branch, hence the "let"
-    let formattedUrl = url;
+    // throw if type is unrecognized
+    if (!mediaPathPrefixes.has(type)) {
+        throw new Error(`Unrecognized media type ${type}`);
+    }
+
+    // prepend media directory to path
+    return joinUrls(mediaPathPrefixes.get(type), url);
+};
+
+// default to `/img/resize`
+const resizeBase = joinUrls(
+    process.env.IMAGE_SERVICE_PATH || '/img/',
+    '/resize/'
+);
+
+const makeOptimizedUrl = (path, width, useFastly) => {
+    const urlObject = new URL(path, window.location.href);
+    const params = new URLSearchParams(urlObject.search);
 
-    if (type) {
-        if (!mediaPathPrefixes.hasOwnProperty(type)) {
-            throw new Error(`Unrecognized media type ${type}`);
+    if (useFastly) {
+        params.set('auto', 'webp');
+        params.set('format', 'pjpg');
+
+        if (width) {
+            params.set('width', width);
         }
-        // "/path/to/jpeg" becomes e.g. "/media/catalog/product/path/to/jpeg"
-        formattedUrl = joinUrls(mediaPathPrefixes[type], url);
+
+        urlObject.search = `?${params}`;
+    } else if (width) {
+        // set pathname, but retain origin
+        urlObject.pathname = `${resizeBase}${width}`;
+
+        // encodeURIComponent would be redundant
+        params.set('url', path);
+
+        urlObject.search = `?${params}`;
     }
 
-    // if a width exists, make a resized URL!
-    return width ? makeResizedUrl(formattedUrl, width) : formattedUrl;
-}
+    return urlObject.href;
+};
 
-// Each combination of type, url, and width is computed only once and cached.
-// storeUrlKey defines the cache key as a function of the arguments.
-const storeUrlKey = (url, { width = '', type = '' } = {}) =>
-    `${type}%%${url}%%${width}`;
+const formatUrl = (url = '', { useFastly, type, width } = {}) =>
+    makeOptimizedUrl(
+        prependMediaPath(url, type),
+        width,
+        useFastly || process.env.USE_FASTLY
+    );
 
-export default memoize(formatUrl, storeUrlKey);
+export default formatUrl;
```

Comment:
Could add memoization back at some point, but it should take `useFastly` into account.

Reply:
Of course, it was the memoization. Should have figured that out.